File: frontend/src/lib/production.ts
import { api } from "./api";
import type { components } from "./api-types";

export type ProductionRecord = components["schemas"]["ProductionOut"];
export type ProductionCreate = components["schemas"]["ProductionCreate"];

export async function fetchProduction(farmId?: string) {
  const res = await api.get<ProductionRecord[]>("/production", { params: farmId ? { farm_id: farmId } : {} });
  return res.data;
}

export async function recordProduction(payload: ProductionCreate) {
  const res = await api.post<ProductionRecord>("/production", payload);
  return res.data;
}

File: backend/app/api/v1/production.py
import uuid
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.core.deps import get_current_user, require_permission
from app.models.user import User
from app.schemas.production import ProductionCreate, ProductionOut
from app.services import production_service
from app.repositories import production_repository

router = APIRouter()


@router.get("", response_model=list[ProductionOut], dependencies=[Depends(get_current_user)])
def list_production(
    farm_id: uuid.UUID | None = None,
    livestock_id: uuid.UUID | None = None,
    product_type: str | None = None,
    db: Session = Depends(get_db),
):
    return production_repository.list_production(db, farm_id, livestock_id, product_type)


@router.post("", response_model=ProductionOut, dependencies=[Depends(require_permission("production.create"))])
def record_production(payload: ProductionCreate, db: Session = Depends(get_db), current_user: User = Depends(get_current_user)):
    return production_service.record_production(db, payload, current_user.id)
